src/nvim/testdir/test_autochdir.vim

```diff
@@ -0,0 +1,17 @@
+" Test 'autochdir' behavior
+
+if !exists("+autochdir")
+  finish
+endif
+
+func Test_set_filename()
+  call test_autochdir()
+  set acd
+  new
+  w samples/Xtest
```

Comment:
This is failing because it's relying on the samples directory to exist.  That will be fixed once my quickfix patches (#5761) are merged.